lib/src/cpp/utils/XMLManager.cpp

```diff
@@ -298,15 +298,23 @@ void XMLManager::create_node(
     reference_node =
             static_cast<xercesc::DOMNode*>(doc->createElement(xercesc::XMLString::transcode(tag_name.c_str())));
 
+    // Add text node to correctly set closing tag indentation
+    std::string node_formatting(LINE_BREAK);
+    for (int i = 0; i < depth_level - 1; i++)
+    {
+        node_formatting += "  ";
```

Comment:
Maybe we can include also a `constexpr static const char* NODE_IDENTATION = "  ";` in the Manager header too.
```suggestion
        node_formatting += NODE_IDENTATION;
```